bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/ui/navigator/CSymbolsContentProvider.java

```diff
@@ -83,10 +87,15 @@ protected void refreshTreeContentFromLS(IFile file) {
 			symbols.cancel(true);
 		}
 
-		final var params = new DocumentSymbolParams(LSPEclipseUtils.toTextDocumentIdentifier(file.getLocationURI()));
-
-		IDocument document = LSPEclipseUtils.getDocument(file);
+		boolean temporaryLoadedDocument = false;
```

Comment:
I see that this seems like a standard practice in LSP4E - ref org.eclipse.lsp4e.LSPEclipseUtils.getExistingDocument(IResource) and a couple of other places - that case uses try/finally so the disconnect call isn't lost and probably should be the same case here?

Reply:
Initially I had the same idea, but then I put all throwable code in a smaller try-catch-statement. This makes it easier to recognize the throwable code parts.

Reply:
That is good, but doesn't stop you from surrounding the whole thing with a try/finally to handle runtime exceptions. e.g. the join on line 117 can throw CancellationException and CompletionException